Source/WebCore/bindings/js/JSDOMAsyncIterator.h

```diff
@@ -139,17 +140,31 @@ template<typename JSWrapper, typename IteratorTraits> class JSDOMAsyncIteratorBa
 
     static void destroy(JSC::JSCell*);
 
+    class IsFinished : public RefCountedAndCanMakeWeakPtr<IsFinished> {
+    public:
+        static Ref<IsFinished> create() { return adoptRef(*new IsFinished); }
+
+        operator bool() { return m_value; }
+        void setAsFinished() { m_value = true; }
```

Comment:
We usually use `markAsFinished()` naming for such things I believe, rather than `setAs..`